File: output/qwen_2.5_72b_instruct/avatar/direct_translation/java/python/failed_translation/atcoder_AGC019_A.py

import sys

Q = int(sys.stdin.readline())
H = int(sys.stdin.readline())
S = int(sys.stdin.readline())
D = int(sys.stdin.readline())
N = int(sys.stdin.readline())

onePrice = min(Q * 4, H * 2, S)
if onePrice <= D / 2:
    print(N * onePrice)
else:
    print((N // 2) * D + (N % 2) * onePrice)

File: output/gpt_4o/avatar/algo_based_translation/python/java/failed_translation/atcoder_AGC019_A.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class atcoder_AGC019_A {

    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

        StringTokenizer st = new StringTokenizer(reader.readLine());
        int q = Integer.parseInt(st.nextToken());
        int h = Integer.parseInt(st.nextToken());
        int s = Integer.parseInt(st.nextToken());
        int d = Integer.parseInt(st.nextToken());

        int n = Integer.parseInt(reader.readLine().trim());

        int best1L = Math.min(q * 4, Math.min(h * 2, s));
        int best2L = Math.min(d, best1L * 2);

        if (n % 2 == 0) {
            System.out.println(best2L * (n / 2));
        } else {
            System.out.println(best2L * (n / 2) + best1L);
        }
    }
}

